src/main/java/org/openrewrite/java/testing/junit6/MinimumJreConditions.java

```diff
@@ -0,0 +1,412 @@
+    private static final String DISABLED_FOR_JRE_RANGE = "org.junit.jupiter.api.condition.DisabledForJreRange";
+    private static final Annotated.Matcher ENABLED_JRE_MATCHER = new Annotated.Matcher("@" + ENABLED_ON_JRE);
+    private static final Annotated.Matcher DISABLED_JRE_MATCHER = new Annotated.Matcher("@" + DISABLED_ON_JRE);
+    private static final Annotated.Matcher ENABLED_JRE_RANGE_MATCHER = new Annotated.Matcher("@" + ENABLED_FOR_JRE_RANGE);
+    private static final Annotated.Matcher DISABLED_JRE_RANGE_MATCHER = new Annotated.Matcher("@" + DISABLED_FOR_JRE_RANGE);
+    private static final List<Annotated.Matcher> TEST_ANNOTATION_MATCHERS = Arrays.asList(
+            new Annotated.Matcher("@org.junit.jupiter.api.Test"),
+            new Annotated.Matcher("@org.junit.jupiter.api.TestFactory"),
+            new Annotated.Matcher("@org.junit.jupiter.api.TestTemplate"),
+            new Annotated.Matcher("@org.junit.jupiter.api.RepeatedTest"),
+            new Annotated.Matcher("@org.junit.jupiter.params.ParameterizedTest")
+    );
+
+    @Option(displayName = "JRE version", description = "The minimum JRE version to use for test conditions.", example = "17")
+    String javaVersion;
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate JUnit JRE conditions";
+    }
+
+    @Override
+    public String getDescription() {
+        return "This recipe will:\n" + " - Remove tests that are only active on JREs that are below the specified version.\n" + " - Adjust ranges to use minimum the specified version.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext ctx) {
+                J.CompilationUnit c = super.visitCompilationUnit(cu, ctx);
+                if (cu != c) {
+                    maybeRemoveImport("org.junit.jupiter.api.Test");
+                    maybeRemoveImport("org.junit.jupiter.api.TestFactory");
+                    maybeRemoveImport("org.junit.jupiter.api.TestTemplate");
+                    maybeRemoveImport("org.junit.jupiter.api.RepeatedTest");
+                    maybeRemoveImport("org.junit.jupiter.params.ParameterizedTest");
+                    maybeRemoveImport(ENABLED_ON_JRE);
+                    maybeRemoveImport(DISABLED_ON_JRE);
+                    maybeRemoveImport(ENABLED_FOR_JRE_RANGE);
+                    maybeRemoveImport(DISABLED_FOR_JRE_RANGE);
+                }
+                return c;
+            }
+
+            @Override
+            public J.@Nullable MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+                boolean isUnitTest = false;
+                Optional<List<String>> enabledOnJre = Optional.empty();
```

Comment:
I think I would drop the Optionals here, as a null check (instead of `.isPresent()`) for this properties be fine as well.